src/plugins/engagement/public/components/iframe_chat/iframe_chat.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useEffect, useRef, useState, CSSProperties } from 'react';
+import { css } from '@emotion/react';
+import { useChat } from '../../services';
+
+type UseChatType =
+  | [false]
+  | [true, string, React.MutableRefObject<HTMLIFrameElement | null>, CSSProperties];
+
+const MESSAGE_READY = 'driftIframeReady';
+const MESSAGE_RESIZE = 'driftIframeResize';
+const MESSAGE_SET_CONTEXT = 'driftSetContext';
+
+const iframeStyle = css`
+  position: fixed;
+  botton: 30px;
+  right: 30px;
+  display: block;
+`;
+
+const getContext = () => {
+  const { location, navigator, innerHeight, innerWidth } = window;
+  const { hash, host, hostname, href, origin, pathname, port, protocol, search } = location;
+  const { language, userAgent } = navigator;
+  const { title, referrer } = document;
```

Comment:
What's the purpose of this function compared to using `window` and `document` directly? To have more control over what's we're effectively sending and avoid potentially leaking things when calling `chatIframe.contentWindow.postMessage`?

Reply:
Just understanding what we have to send to the frame.  This is a refactored bit of another draft PR from the Journey team, and, since I don't have insight into the messaging API, splitting it out like this seemed like a good way to keep track of precisely what we need to send, (and for testing purposes).